movieland_persistence_jdbc/src/main/java/com/solovey/movieland/dao/jdbc/JdbcReportingDao.java

```diff
@@ -0,0 +1,166 @@
+package com.solovey.movieland.dao.jdbc;
+
+
+import com.solovey.movieland.dao.ReportingDao;
+import com.solovey.movieland.dao.jdbc.mapper.ReportMovieRowMapper;
+import com.solovey.movieland.dao.jdbc.mapper.ReportTopUserRowMapper;
+import com.solovey.movieland.entity.reporting.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class JdbcReportingDao implements ReportingDao {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private static final ReportMovieRowMapper REPORT_MOVIE_ROW_MAPPER = new ReportMovieRowMapper();
+    private static final ReportTopUserRowMapper REPORT_TOP_USER_ROW_MAPPER = new ReportTopUserRowMapper();
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    @Autowired
+    private QueryGenerator queryGenerator;
+
+    @Autowired
+    private String getMoviesForReportSql;
+
+    @Autowired
+    private String saveReportMetadataSql;
+
+    @Autowired
+    private String getReportsMetadataSql;
+
+    @Autowired
+    private String getMaxReportIdSql;
+
+    @Autowired
+    private String removeReportMetadataSql;
+
+    @Autowired
+    private String getTopUsersSql;
+
+    @Override
+    public List<ReportMovie> getMoviesForReport(Report report) {
+
+        log.debug("Start getting movies for report from DB");
+        long startTime = System.currentTimeMillis();
+
+        Date inDateFrom;
+        Date inDateTo;
+        List<ReportMovie> movieList;
+        if (report.getReportType() == ReportType.ADDED_DURING_PERIOD) {
+            inDateFrom = Date.valueOf(report.getDateFrom());
+            inDateTo = Date.valueOf(report.getDateTo());
+            movieList = jdbcTemplate.query(queryGenerator.buildMoviesReportQuery(getMoviesForReportSql, report.getReportType()),
+                    REPORT_MOVIE_ROW_MAPPER, inDateFrom, inDateTo);
+        } else {
+            movieList = jdbcTemplate.query(queryGenerator.buildMoviesReportQuery(getMoviesForReportSql, report.getReportType()),
+                    REPORT_MOVIE_ROW_MAPPER);
+        }
+
+
+        log.debug("Finish getting movies for report from DB. It took {} ms", System.currentTimeMillis() - startTime);
+        return movieList;
+    }
```

Comment:
From my perspective each method in Jdbc layer should have ability to extract only one data set. It will give us ability to move IF condition logic on service layer (Fix not required)